### PR TITLE
Match events to JSON schema

### PR DESCRIPTION
**Pull request by NickGrantONS**
# Motivation and Context
Our events didn't conform to any interface contract. We've now got JSON schemas which form the interface contract, so we should make changes to meet that spec.

# What has changed
Complied with interface spec: https://github.com/ONSdigital/ssdc-shared-events/tree/main/event_dictionary/v0.2_RELEASE

Still a bit of work remaining though. There will be follow up tickets.

# How to test?
Build Case Processor, Case API, Support Tool and Response Operations, then start everything up in docker and run the acceptance tests. There should be zero regression.

# Links
Trello: https://trello.com/c/rPyzdFeY

### CODE REVIEW

**Comment by NickGrantONS** [comment]
[Comply with Events JSON Schema v0.2 RELEASE (8)](https://trello.com/c/rPyzdFeY/2666-comply-with-events-json-schema-v02-release-8)

**Review by LeoKHoward — APPROVED** [review]
All built fine and ATs passed

File: README.md [hunk 1]
```diff
@@ -29,13 +29,10 @@ Just build the image
...
 
-## Events
+## Internal Events
 
 ### Sample Load
-Queue name: `supportTool.caseProcessor.sample`
-Exchange: N/A
-Routing key: N/A
-
+Topic: `rm-internal-telephone-capture`
```
Comment (ryangrundy7) [inline README.md:35]: Think this should be `event_case-update` instead?

File: src/main/java/uk/gov/ons/ssdc/caseprocessor/model/dto/PayloadDTO.java [hunk 2]
```diff
@@ -7,13 +7,15 @@
...
   private RefusalDTO refusal;
-  private CollectionCase collectionCase;
-  private UacDTO uac;
-  private InvalidAddress invalidAddress;
-  private FulfilmentDTO fulfilment;
+  private CaseUpdateDTO caseUpdate;
+  private UacUpdateDTO uacUpdate;
+  private InvalidCase invalidCase;
+  private PrintFulfilmentDTO printFulfilment;
   private TelephoneCaptureDTO telephoneCapture;
```
Comment (ryangrundy7) [inline src/main/java/uk/gov/ons/ssdc/caseprocessor/model/dto/PayloadDTO.java:16]: I can't see anything about telephone capture in the event dictionary, was it missed out?

File: src/main/java/uk/gov/ons/ssdc/caseprocessor/model/dto/UacUpdateDTO.java [hunk 3]
```diff
@@ -7,10 +7,9 @@
 
 @Data
 @JsonInclude(Include.NON_NULL)
-public class UacDTO {
+public class UacUpdateDTO {
```
Comment (ryangrundy7) [inline src/main/java/uk/gov/ons/ssdc/caseprocessor/model/dto/UacUpdateDTO.java:10]: Missing uacHash from here?

File: src/test/java/uk/gov/ons/ssdc/caseprocessor/messaging/ManagedMessageRecovererTest.java [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    ExceptionReportResponse exceptionReportResponse = new ExceptionReportResponse();
+    RetryContext retryContext = testSetupTestRecover(exceptionReportResponse);
+
+    // When
+    underTest.recover(retryContext);
+
+    // Then
+    verify(exceptionManagerClient)
+        .reportException(
+            eq("90f56b5b3ffe9558a546af25a7256da4b2761864575f9d59c81b70629023465b"),
```
Comment (AdamHawtin) [inline src/test/java/uk/gov/ons/ssdc/caseprocessor/messaging/ManagedMessageRecovererTest.java:52]: Lots of copy paste of this same hash string 🎨

**Review by AdamHawtin — APPROVED** [review]
Building and passing ATs 👍

File: src/main/java/uk/gov/ons/ssdc/caseprocessor/model/dto/PayloadDTO.java (hunk 2 above)
Comment (AdamHawtin) [reply src/main/java/uk/gov/ons/ssdc/caseprocessor/model/dto/PayloadDTO.java:16]: It's an internal only event, not something we're accepting from outside systems.

File: src/main/java/uk/gov/ons/ssdc/caseprocessor/model/dto/UacUpdateDTO.java (hunk 3 above)
Comment (AdamHawtin) [reply src/main/java/uk/gov/ons/ssdc/caseprocessor/model/dto/UacUpdateDTO.java:10]: Those are being handled on a separate ticket, they need some more thought and/or changes to other features.

File: README.md (hunk 1 above)
Comment (NickGrantONS) [reply README.md]: fixed

**Review by ryangrundy7 — APPROVED** [review]

File: src/test/java/uk/gov/ons/ssdc/caseprocessor/messaging/ManagedMessageRecovererTest.java (hunk 4 above)
Comment (NickGrantONS) [reply src/test/java/uk/gov/ons/ssdc/caseprocessor/messaging/ManagedMessageRecovererTest.java]: fixed

**Pull request merged**